### DISCUSSION TITLE
Documentation?

### DISCUSSION THREAD

**Discussion by Vosburgh** [bluez/bluez#951] [General]
# Where is the Documentation?

I'm trying to configure some bluez properties on Nix but there doesn't seem to be documentation anywhere? Or at the least it's very hidden away. Maybe it's just an ancient API passed down from master to apprentice via word of mouth. 

Anyway, where can we read documentation on how what properties are available that doesn't disrespect everyone's time with searching through obscure tomes?

It's confusing when you go through many attempted extraConfig changes to see if your BT headset will stop changing to HSP mode, only for every solution to fail. Many others and myself would not have this wasted time if there were publicly available documentation.

**Comment by Vudentz** [comment]
> # Where is the Documentation?
> 
> I'm trying to configure some bluez properties on Nix but there doesn't seem to be documentation anywhere? Or at the least it's very hidden away. Maybe it's just an ancient API passed down from master to apprentice via word of mouth.
> 
> Anyway, where can we read documentation on how what properties are available that doesn't disrespect everyone's time with searching through obscure tomes?
> 
> It's confusing when you go through many attempted extraConfig changes to see if your BT headset will stop changing to HSP mode, only for every solution to fail. Many others and myself would not have this wasted time if there were publicly available documentation.

HSP/HFP is under PulseAudio/Pipewire control, it is implemented via Profile interface:

https://github.com/bluez/bluez/blob/master/doc/org.bluez.Profile.rst

Anyway sounds like you are having problems with audio routing defaulting to HSP, perhaps look into pipewire docs e.g:

https://docs.pipewire.org/page_man_pipewire-devices_7.html#device-param__bluetooth_properties

**Comment by stokito** [comment]
Here on the GitHub in the project description is mentioned the site [bluez.github.io/bluez/](https://bluez.github.io/bluez/) from the GitHub pages but it doesn't work. Maybe you can fix it and put the documentation there?
